### DISCUSSION TITLE
Does Autobahn-JS support cryptosign authentication?

### DISCUSSION THREAD

**Discussion by waterbug** [crossbario/crossbar#2072] [Q&A]
My use case is that I have a mature python desktop application that uses cryptosign authentication to connect to crossbar and access a repository service, and I would like to implement a similar (but simpler) browser app using autobahn-js.  I already have many users set up with public/private key pairs so obviously I would prefer to avoid implementing another authentication mechanism for a web client.  If cryptosign auth is not available with autobahn-js, do you have a suggestion for the "most compatible" alternative -- e.g. would it be possible to do some other autobahn-js supported auth based on the user's key?

**Comment by oberstet** [comment]
cryptosign is available in AutobahnJS - and it is actively used/supported. so I wouldn't change anything in your app ... use cryptosign both for desktop and web. in fact, being able to do so (transparently use native/desktop and web frontends) was a main reason/goal with WAMP .. and cryptosign is very good for that rgd the authentication aspect.
